File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [0];
    let arr=progresses.map((v,i)=>Math.ceil((100-v)/speeds[i]));
    let maxDay=arr[0];
    
    for(let i=0, j=0; i<arr.length;i++){
        if(maxDay>=arr[i]){
            answer[j]+=1;
        }else{
            maxDay=arr[i];
            answer[++j]=1;
        }
    }
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer=new LinkedList<>();
        int n=progresses.length;
                
        Queue<Integer> q=new LinkedList<>();
        
        for(int i=0; i<n; i++){
            int spendTime=100-progresses[i];
            if(spendTime%speeds[i]!=0){
                spendTime=spendTime/speeds[i]+1;
            }else{
                spendTime/=speeds[i];
            }
            q.add(spendTime);
        }
        
        while(!q.isEmpty()){
            int cur=q.poll();
            int oneDaySuccess=1;
            while(!q.isEmpty()&&q.peek()<=cur){
               oneDaySuccess++;
               q.poll();
            }
            answer.add(oneDaySuccess);
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}